src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp

```diff
@@ -44,8 +46,33 @@ class TBuildSingleAgentListInfoContext final : public AbstractHandler<std::share
      */
     std::shared_ptr<TScanContext> handleRequest(std::shared_ptr<TScanContext> data) override
     {
-        data->m_agents.push_back(
-            {data->agentId().data(), data->agentName().data(), data->agentVersion().data(), data->agentIp().data()});
+        nlohmann::json response;
+
+        try
+        {
+            // Execute query
+            TSocketDBWrapper::instance().query(
+                WazuhDBQueryBuilder::builder()
+                    .globalGetCommand(std::string("agent-info ") + data->agentId().data())
+                    .build(),
+                response);
+        }
+        // LCOV_EXCL_START
+        catch (const std::exception& e)
+        {
+            logDebug2(WM_VULNSCAN_LOGTAG, "Error executing query to fetch global agent data. Reason: %s.", e.what());
+            throw WdbDataException("Error executing query to fetch global agent data", data->agentId().data());
+        }
+        // LCOV_EXCL_STOP
```

Comment:
No need for the lcov flags
```suggestion
        catch (const std::exception& e)
        {
            logDebug2(WM_VULNSCAN_LOGTAG, "Error executing query to fetch global agent data. Reason: %s.", e.what());
            throw WdbDataException("Error executing query to fetch global agent data", data->agentId().data());
        }
```